src/System.Management.Automation/DscSupport/JsonDscClassCache.cs

```diff
@@ -0,0 +1,3249 @@
+
+            // previous version of this code was the only place that referenced CimSuperClass
+            // so to simplify things we just check superclass to be OMI_BaseResource
+            // with assumption that current code will not work for multi-level class inheritance (which is never used in practice according to DSC team)
+            // this simplification allows us to avoid linking objects together using CimSuperClass field during deserialization from json schema
+            if ((!string.IsNullOrEmpty(cimClass.SuperClassName)) && string.Equals("OMI_BaseResource", cimClass.SuperClassName, StringComparison.OrdinalIgnoreCase))
+            {
+                isResourceType = true;
+            }
+
+            // If it's a resource type, then a resource name is required.
+            keyword.NameMode = isResourceType ? DynamicKeywordNameMode.NameRequired : DynamicKeywordNameMode.NoName;
+
+            // Add the settable properties to the keyword object
+            if (cimClass.ClassProperties != null)
+            {
+                foreach (var prop in cimClass.ClassProperties)
+                {
+                    // If the property has the Read qualifier, skip it.
+                    if (string.Equals(prop.Qualifiers?.Read?.ToString(), "True", StringComparison.OrdinalIgnoreCase))
+                    {
+                        continue;
+                    }
+
+                    // If it's one of our magic properties, skip it
+                    if (IsMagicProperty(prop.Name))
+                    {
+                        continue;
+                    }
+
+                    if (runAsBehavior == DSCResourceRunAsCredential.NotSupported)
+                    {
+                        if (string.Equals(prop.Name, "PsDscRunAsCredential", StringComparison.OrdinalIgnoreCase))
+                        {
+                            // skip adding PsDscRunAsCredential to the dynamic word for the dsc resource.
+                            continue;
+                        }
+                    }
+
+                    // If it's one of our reserved properties, save it for error reporting
+                    if (System.Text.RegularExpressions.Regex.Match(prop.Name, reservedProperties, System.Text.RegularExpressions.RegexOptions.IgnoreCase).Success)
+                    {
+                        keyword.HasReservedProperties = true;
+                        continue;
+                    }
+
+                    // Otherwise, add it to the Keyword List.
+                    var keyProp = new System.Management.Automation.Language.DynamicKeywordProperty();
+                    keyProp.Name = prop.Name;
+
+                    // Copy the type name string. If it's an embedded instance, need to grab it from the ReferenceClassName
+                    bool referenceClassNameIsNullOrEmpty = string.IsNullOrEmpty(prop.ReferenceClassName);
+                    if (prop.CimType == "Instance" && !referenceClassNameIsNullOrEmpty)
```

Comment:
CimType has Reference as well which also represents Instance..
Not sure if it's used any where..

Reply:
Original code didn't care about those, so probably it is fine as is.